stream_in.c

```diff
@@ -390,9 +390,46 @@ int in_get_active_microphones(const struct audio_stream_in *stream,
                               struct audio_microphone_characteristic_t *mic_array,
                               size_t *mic_count)
 {
+    x_stream_in_t *xin = (x_stream_in_t*)stream;
+
+    if ((xin == NULL) || (mic_array == NULL) || (mic_count == NULL)) {
+        return -EINVAL;
+    }
+
+    pthread_mutex_lock(&xin->lock);
     LOG_FN_NAME_WITH_ARGS("(%p)", stream);
-    /* TODO To implement */
-    return -ENOSYS;
+
+    *mic_count = 1;
```

Comment:
Do we able to have only one active mic? Comment here could be be useful.

Reply:
Yes, from my point of view. This function is called for open input stream so we always have one mic.Comment added.